File: app/src/main/java/com/pluu/support/tstore/TStorerWeekApi.java
package com.pluu.support.tstore;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import com.pluu.support.impl.AbstractWeekApi;
import com.pluu.support.impl.ServiceConst;
import com.pluu.webtoon.R;
import com.pluu.webtoon.item.Status;
import com.pluu.webtoon.item.WebToonInfo;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

/**
 * TStore 웹툰 Week Api
 * Created by PLUUSYSTEM-NEW on 2015-10-31.
 */
public class TStorerWeekApi extends AbstractWeekApi {

	private static final String[] TITLE = new String[]{"월", "화", "수", "목", "금", "토", "T툰"};
	private final String WEEKLY_URL = "http://m.tstore.co.kr/mobilepoc/webtoon/weekdayList.omp?weekday=";
	private final Pattern URL_PATTERN = Pattern.compile("(?<=goInnerUrlDetail\\(\\\\\\').+(?=\\\\'\\)'\\);)");
	private final Pattern ID_PATTERN = Pattern.compile("(?<=prodId=).+(?=&)");

	private int currentPos;

	public TStorerWeekApi() {
		super(TITLE);
	}

	@Override
	public ServiceConst.NAV_ITEM getNaviItem() {
		return ServiceConst.NAV_ITEM.T_STORE;
	}

	@Override
	public List<WebToonInfo> parseMain(int position) {
		this.currentPos = position;

		ArrayList<WebToonInfo> list = new ArrayList<>();

		try {
			String response = requestApi();
			Document doc = Jsoup.parse(response);

			WebToonInfo item;
			String href;

			Matcher matcher, matcher2;
			Elements bottom = doc.select("#weekToon0" + (position + 1) + " ul li a");

			for (Element a : bottom) {
				href = a.attr("href");
				matcher = URL_PATTERN.matcher(href);
				if (!matcher.find()) {
					continue;
				}
				matcher2 = ID_PATTERN.matcher(matcher.group());
				if (!matcher2.find()) {
					continue;
				}

				item = new WebToonInfo(matcher2.group());
				item.setTitle(a.select(".detail dl dt").text());
				item.setImage(a.select(".thum img").last().attr("src"));
				item.setWriter(a.select(".txt").text());
//				item.setRate(a.select(".cRed1").text());
				item.setUpdateDate(a.select(".grade strong").text());

				if (!a.select(".new-up").isEmpty()) {
					// 최근 업데이트
					item.setStatus(Status.UPDATE);
				}

				list.add(item);
			}
		} catch (Exception e) {
			e.printStackTrace();
		}

		return list;
	}

	@Override
	public String getMethod() {
		return POST;
	}

	@Override
	public String getUrl() {
		return WEEKLY_URL + (currentPos + 1) ;
	}
}

File: app/src/main/java/com/pluu/support/tstore/TStorerWeekApi.kt
package com.pluu.support.tstore

import android.content.Context
import com.pluu.support.impl.AbstractWeekApi
import com.pluu.support.impl.NAV_ITEM
import com.pluu.support.impl.NetworkSupportApi
import com.pluu.webtoon.item.Status
import com.pluu.webtoon.item.WebToonInfo
import org.jsoup.Jsoup

/**
 * TStore 웹툰 Week Api
 * Created by pluu on 2017-04-27.
 */
class TStorerWeekApi(context: Context) : AbstractWeekApi(context, TStorerWeekApi.TITLE) {
    private val ID_PATTERN = "(?<=prodId=).+(?=&)".toRegex()
    private val IMG_PATTERN = "(?<=url\\(').+(?='\\);)".toRegex()
    private val DATE_PATTERN = "\\d{4}.\\d{2}.\\d{2}".toRegex()

    private var currentPos: Int = 0

    override val naviItem: NAV_ITEM
        get() = NAV_ITEM.T_STORE

    override fun parseMain(position: Int): List<WebToonInfo> {
        this.currentPos = position

        val doc = try {
            Jsoup.parse(requestApi())
        } catch (e: Exception) {
            e.printStackTrace()
            return emptyList()
        }

        val list = mutableListOf<WebToonInfo>()
        doc.select("#weekToon0" + (position + 1) + " ul li a").forEach {
            ID_PATTERN.find(it.attr("href"))?.apply {
                WebToonInfo(value).apply {
                    title = it.select(".list-item-text-title").text()
                    image = IMG_PATTERN.find(it.select("span[class=list-thumbnail-pic ebook-lazy]").attr("style"))?.value
                    writer = it.select(".list-item-text-like-info").last()?.children()?.last()?.text()
                    updateDate = DATE_PATTERN.find(it.select("span[class=list-item-text-date list-item-text-point]").text())?.value

                    if (it.select("i[class=icon-type-ktoon icon-badge-update]").isNotEmpty()) {
                        // 최근 업데이트
                        status = Status.UPDATE
                    }
                    list.add(this)
                }
            }
        }
        return list
    }

    override val method: String
        get() = NetworkSupportApi.POST

    override val url: String
        get() = "http://m.tstore.co.kr/mobilepoc/webtoon/weekdayList.omp?weekday=${currentPos + 1}"

    companion object {
        private val TITLE = arrayOf("월", "화", "수", "목", "금", "토", "T툰")
    }
}
